vscode/src/services/HistoryChat.ts

```diff
@@ -0,0 +1,145 @@
+import * as vscode from 'vscode'
+
+import { getChatPanelTitle } from '../chat/chat-view/chat-helpers'
+import { chatHistory } from '../chat/chat-view/ChatHistoryManager'
+
+import { CodySidebarTreeItem } from './treeViewItems'
+
+interface GroupedChats {
+    [groupName: string]: CodySidebarTreeItem[]
+}
+
+interface HistoryItems {
```

Comment:
This be history *item*, singular.
```suggestion
interface HistoryItem {
```